eva-server/src/main/java/uk/ac/ebi/eva/server/ws/GeneWSServer.java

```diff
@@ -19,103 +19,84 @@
+
+        if (species.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setQueryResponse("Please specify a species");
         }
-        
-        queryOptions.put(VariantDBAdaptor.SORT, new BasicDBObject("chr", 1).append("start", 1));
-        queryOptions.put(VariantDBAdaptor.GENE, String.join(",", geneIds));
-        
-        return setQueryResponse(variantMongoDbAdaptor.getAllVariants(queryOptions));
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
+
+        List<VariantEntityRepositoryFilter> filters =
+                new FilterBuilder().getVariantEntityRepositoryFilters(maf, polyphenScore, siftScore, studies,
+                                                                      consequenceType);
+
+        List<VariantEntity> variantEntities =
+                variantEntityRepository.findByGenesAndComplexFilters(geneIds, filters,
+                                                                     Utils.getPageRequest(queryOptions));
+        Long numTotalResults = variantEntityRepository.countByGenesAndComplexFilters(geneIds, filters);
+
+        QueryResult<VariantEntity> queryResult = buildQueryResult(variantEntities, numTotalResults);
+        return setQueryResponse(queryResult);
     }
 
     @RequestMapping(value = "/{geneIds}/variants", method = RequestMethod.POST)
     public QueryResponse getVariantsByGenePOST(@PathVariable("geneIds") List<String> geneIds,
                                                @RequestParam(name = "species") String species,
-                                               @RequestParam(name = "studies", required = false) String studies,
+                                               @RequestParam(name = "studies", required = false) List<String> studies,
                                                @RequestParam(name = "annot-ct", required = false) List<String> consequenceType,
                                                @RequestParam(name = "maf", defaultValue = "") String maf,
                                                @RequestParam(name = "polyphen", defaultValue = "") String polyphenScore,
                                                @RequestParam(name = "sift", defaultValue = "") String siftScore,
-                                               @RequestParam(name = "ref", defaultValue = "") String reference,
-                                               @RequestParam(name = "alt", defaultValue = "") String alternate,
-                                               @RequestParam(name = "miss_alleles", defaultValue = "") String missingAlleles,
-                                               @RequestParam(name = "miss_gts", defaultValue = "") String missingGenotypes)
+                                               HttpServletResponse response)
             throws UnknownHostException, IllegalOpenCGACredentialsException, IOException {
```

Comment:
Some of these exceptions, in particular `IllegalOpenCGACredentialsException`, may not be thrown anymore.